draft-ietf-privacypass-architecture.md

```diff
@@ -107,162 +82,251 @@ of all participating entities.
-to track client redemption patterns over time. Clients cannot check
-whether servers monitor such identifying information. Thus, clients
-SHOULD minimize or remove identifying information where possible, e.g.,
-by using anonymity-preserving tools such as Tor to interact with
-servers.
-
-# Key management and discovery {#key-mgmt}
-
-The key material and protocol configuration that a server uses to issue
+{: #fig-issuance title="Issuance Overview"}
+
+Each issuance protocol may be different, e.g., in the number and types of
+participants, underlying cryptographic constructions used when issuing tokens,
+and even privacy properties.
+
+Clients initiate the Token issuance protocol using the challenge, a randomly
+generated nonce, and public key for the Issuer. The Token issuance protocol
+itself can be any interactive protocol between Client, Issuer, or other
+parties that produces a valid authenticator over the Client's input, subject
+to the following security requirements.
+
+1. Unconditional input secrecy. The issuance protocol MUST NOT reveal anything
+about the Client's private input, including the challenge and nonce. The
+issuance protocol can reveal the Issuer public key for the purposes of
+determining which private key to use in producing the issuance protocol. A
+result of this property is that the redemption flow is unlinkable from the
+issuance flow.
+1. One-more forgery security. The issuance protocol MUST NOT allow malicious
+Clients to forge tokens without interacting with the Issuer directly.
+1. Concurrent security. The issuance protocol MUST be safe to run concurrently
+with arbitrarily many Clients.
+
+Each Issuance protocol MUST come with a detailed analysis of the privacy impacts
+of the protocol, why these impacts are justified, and guidelines on changes to
+the parametrization in {{parametrization}}.
+
+The mechanism by which clients obtain the Issuer public key is not specified.
+Clients may be configured with this key or they may discover it via some other
+form. See {{?CONSISTENCY=I-D.wood-key-consistency}}.
+
+Depending on the use case, issuance may require some form of Client
+anonymization service similar to an IP-hiding proxy so that Issuers cannot
+learn information about Clients. This can be provided by an explicit
+participant in the issuance protocol, or it can be provided via external means,
+e.g., through the use of an IP-hiding proxy service like Tor. In general,
+Clients SHOULD minimize or remove identifying information where possible when
+invoking the issuance protocol.
+
+Issuers MUST NOT issue tokens for Clients through untrusted Attesters. This is
+important because the Attester's role is to vouch for trust in
+privacy-sensitive Client information, such as account identifiers or IP address
+information, to the Issuer. Tokens produced by an Issuer that admits issuance
+for any type of attestation cannot be relied on for any specific property.
+See {{attester-role}} for more details.
+
+### Attester Role
```

Comment:
Maybe later on, but should it be explicitly called out that some forms of the protocol might collapse this into another role?